public/js/actions/subscriptions.js

```diff
@@ -33,43 +33,68 @@ export function getUserSubscriptions(jquery=$) {
 }
 
 
-export function createSubscription(braintreeToken, productId,
-                                   creditCard, jquery=$,
+export function createSubscription(productId, payMethod,
+                                   braintreeToken, jquery=$,
                                    BraintreeClient=braintree.api.Client) {
   return dispatch => {
 
-    var client = new BraintreeClient({
-      clientToken: braintreeToken,
-    });
+    function requestSub(opts={}) {
+      var data = {
+        plan_id: productId,
+      };
+      data.pay_method_uri = opts.payMethodUri;
+      data.pay_method_nonce = opts.nonce;
 
-    client.tokenizeCard({
-      number: creditCard.number,
-      expirationDate: creditCard.expiration,
-      cvv: creditCard.cvv,
-    }, (err, nonce) => {
-      if (err) {
-        console.error('Braintree tokenization error:', err);
-        dispatch(appActions.error('Braintree tokenization error'));
-      } else {
-        jquery.ajax({
-          data: {
-            pay_method_nonce: nonce,
-            plan_id: productId,
-          },
-          url: '/api/braintree/subscriptions/',
-          method: 'post',
-          dataType: 'json',
-        }).then(() => {
-          console.log('Successfully subscribed + completed payment');
-          dispatch(transactionActions.complete());
-        }).fail($xhr => {
+      return jquery.ajax({
+        data: data,
+        url: '/api/braintree/subscriptions/',
+        method: 'post',
+        dataType: 'json',
+      }).then(() => {
+        console.log('Successfully subscribed + completed payment');
+        dispatch(transactionActions.complete());
+      }).fail($xhr => {
+        if (data.pay_method_nonce) {
           dispatch({
             type: actionTypes.CREDIT_CARD_SUBMISSION_ERRORS,
             apiErrorResult: $xhr.responseJSON,
           });
-        });
+        } else {
+          dispatch(appActions.error('Subscription creation failed'));
+        }
```

Comment:
I don't think this if/else is in the right place. I think this should always dispatch CREDIT_CARD_SUBMISSION_ERRORS and the *receiver* of that data should decide how to handle it.